DOL.WHD.Section14c.Api/Controllers/AccountController.cs

```diff
@@ -96,75 +228,208 @@ public ApplicationUserManager UserManager
+            {
+                user.Email = model.Email;
+                user.UserName = model.Email;
+                var userUpdated = UserManager.Update(user);
+
+                if (!userUpdated.Succeeded)
                 {
-                    ModelState.AddModelError("ReCaptchaResponse", new Exception("Unable to validate reCaptcha Response"));
-                    return BadRequest(ModelState);
+                    return GetErrorResult(userUpdated);
                 }
             }
 
-            // Add User
-            var now = DateTime.UtcNow;
-            var user = new ApplicationUser() { UserName = model.Email, Email = model.Email };
-            user.Organizations.Add(new OrganizationMembership { EIN = model.EIN, IsAdmin = true, CreatedAt = now, LastModifiedAt = now, CreatedBy_Id = user.Id, LastModifiedBy_Id = user.Id });
-
-            IdentityResult result = await UserManager.CreateAsync(user, model.Password);
+            // Add Roles
+            foreach (var role in model.Roles)
+            {
+                if (user.Roles.All(x => x.Role.Name != role.Name))
+                {
+                    var addResult = UserManager.AddToRole(user.Id, role.Name);
+                    if (!addResult.Succeeded)
+                    {
+                        return GetErrorResult(addResult);
+                    }
+                }
+            }
 
-            if (!result.Succeeded)
+            // Remove
+            foreach (var role in user.Roles.ToList())
             {
-                return GetErrorResult(result);
+                if (model.Roles.All(x => x.Name != role.Role.Name))
+                {
+                    var removeResult = UserManager.RemoveFromRole(user.Id, role.Role.Name);
+                    if (!removeResult.Succeeded)
+                    {
+                        return GetErrorResult(removeResult);
+                    }
+                }
             }
 
             return Ok();
         }
+        #endregion
+
+        //// POST api/Account/RemoveLogin
+        //[Route("RemoveLogin")]
+        //public async Task<IHttpActionResult> RemoveLogin(RemoveLoginBindingModel model)
+        //{
+        //    if (!ModelState.IsValid)
+        //    {
+        //        return BadRequest(ModelState);
+        //    }
+
+        //    IdentityResult result;
+
+        //    if (model.LoginProvider == LocalLoginProvider)
+        //    {
+        //        result = await UserManager.RemovePasswordAsync(User.Identity.GetUserId());
+        //    }
+        //    else
+        //    {
+        //        result = await UserManager.RemoveLoginAsync(User.Identity.GetUserId(),
+        //            new UserLoginInfo(model.LoginProvider, model.ProviderKey));
+        //    }
+
+        //    if (!result.Succeeded)
+        //    {
+        //        return GetErrorResult(result);
+        //    }
+
+        //    return Ok();
+        //}
```

Comment:
Yea, it is framework code and gone :)

https://github.com/AppliedIS/dol-whd-14c/pull/90/commits/03dd22da59de844ee7b17302302423ecd20a7289